### DISCUSSION TITLE
train.py takes too long when i run locally on jupyter lab (45-46sec per epoch) but faster on colab(3-4sec per epoch)

### DISCUSSION THREAD

**Discussion by SuryanshuBasa** [mrdbourke/pytorch-deep-learning#879] [Q&A] [answered]
While coding the assignment on 05_going_modular locally, the train.py script whej running from CLI takes up 45sec per epoch where as its significantly faster on colab. I have an RTX 3060 6GB. The functions for individual .py scripts are the same form the course. Looking for any help
Thank you :)
<img width="839" alt="Screenshot 2024-03-27 211600" src="https://github.com/mrdbourke/pytorch-deep-learning/assets/98867431/d5e19dd1-66f1-4664-aa81-7516468916ff">
<img width="538" alt="Screenshot 2024-03-27 211628" src="https://github.com/mrdbourke/pytorch-deep-learning/assets/98867431/28f34ae6-7faa-4cd7-ba72-5b19dd5b3969">

**Comment by LuluW8071** [comment, marked as answer]
Could be due to many reasons:

| Feature            | Google Colab                                           | Your Device                       |
|--------------------|--------------------------------------------------------|--------------------------------------------------|
| CPU                | Intel Xeon [Large Cache Memory]                                       | Desktop chip (e.g., Intel Core or AMD Ryzen)    |
| GPU                | Tesla T4 (12GB VRAM)                                   | NVIDIA RTX 3060 (6GB VRAM)                      |
| Software Environment | Linux                                                 | Windows                                          |
| I/O Performance    | Potentially faster due to access to Google Drive and optimized cloud storage solutions | Depends on local storage and network setup        |

Maybe in your case one solution would be increase `num_workers` in your dataloader script to half of your cpu threads and run the train script again. Also remember to check task manager to see other applications resource consumption. Maybe that is slowing down your training progress.

**Reply by SuryanshuBasa** [reply]
Thanks for the info, the table makes it clear :)
